### PR TITLE
Fetch teams

### PR DESCRIPTION
**Pull request by IdoDePro**

### CODE REVIEW

File: lib/views/common/fetch_functions/climb_enum.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  failed("Failed"),
+  climbed("Climbed"),
+  buddyClimbed("Buddy Climbed");
+
+  const Climb(final String title);
+}
+
+Climb climbTitleToEnum(final String title) {
+  switch (title) {
+    case "No Attempt":
```
Comment (itamarsch) [inline lib/views/common/fetch_functions/climb_enum.dart:12]:
```dart
Climb
.values
.where((climbOption) => climbOption.title == title).singleOrNull ?? throw Exception("Invalid title $title")
```

File: lib/views/common/fetch_functions/avg_technical_data.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class AvgData {
```
Comment (itamarsch) [inline lib/views/common/fetch_functions/avg_technical_data.dart:1]: I prefer that you made a `MeanData` class or at least add one

File: lib/views/common/fetch_functions/single-multiple_teams/fetch_teams.dart [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                )
+                .toList(),
+            specificMatches:
+                specificMatchesTables.map(SpecificMatchData.parse).toList(),
+            summaryData: SpecificSummaryData.parse(specificSummaryTable),
+            lightTeam: team,
+          );
+        }),
+        (final TeamData team1, final TeamData team2) =>
+            team1.lightTeam.id.compareTo(team2.lightTeam.id),
```
Comment (itamarsch) [inline lib/views/common/fetch_functions/single-multiple_teams/fetch_teams.dart:176]: I prefer the team number

File: lib/views/common/fetch_functions/single-multiple_teams/fetch_teams.dart [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+          final dynamic pitTable = teamTable["pit"];
+          final dynamic specificSummaryTable = teamTable["specific_summary"];
+          return TeamData(
+            avgData: AvgData.parse(avgTable),
+            technicalMatches:
+                technicalMatchesTable.map(TechnicalMatchData.parse).toList(),
+            pitData: PitData.parse(pitTable),
+            faultEntrys: (teamTable["faults"] as List<dynamic>)
+                .map(
+                  (final dynamic fault) => FaultEntry(
```
Comment (itamarsch) [inline lib/views/common/fetch_functions/single-multiple_teams/fetch_teams.dart:159]: Why not `FaultEntry.parse`?

**Review by itamarsch — CHANGES_REQUESTED** [review]

File: lib/views/common/fetch_functions/avg_technical_data.dart (hunk 2 above)
Comment (IdoDePro) [reply lib/views/common/fetch_functions/avg_technical_data.dart:1]: median...

**Review by itamarsch — APPROVED** [review]

**Pull request merged**